src/uct/cuda/cuda_copy/cuda_copy_iface.h

```diff
@@ -48,23 +49,30 @@ typedef struct uct_cuda_copy_queue_desc {
     ucs_queue_elem_t            queue;
 } uct_cuda_copy_queue_desc_t;
 
+typedef struct uct_cuda_copy_per_ctx_rsc {
+    CUcontext                   cuda_ctx;
+    unsigned long long          ctx_id;
```

Comment:
Please remove `ctx_id` filed.

Reply:
@rakhmets But we use this as the key to get the right context from the hashmap. Are you suggesting that we use a different key?

Reply:
No, I suggest using ctx_id as key, without storing to a value.